src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -1043,6 +1043,80 @@ public void ExceptionWhenTryingToBindToByteArray()
                 exception.Message);
         }
 
+        [Fact]
```

Comment:
>Fact [](http://example.com/codeflow?start=9&length=4)

do we have a test case like the case mentioned in [https://github.com/lenraven/BindPropertyExceptionSample/tree/master/BindPropertyExceptionSample](https://github.com/lenraven/BindPropertyExceptionSample/tree/master/BindPropertyExceptionSample)?

Reply:
I tried to test that with https://github.com/dotnet/runtime/pull/66723/files#diff-22b129d6955920524411aeb76c26d711b280b02f9b44bf043c0b9ffa22aed564R1166 . Always throwing is the "harder" case compared to conditionally throwing; in both cases the key is not to read the property value if we don't need to.

If you wanted, I could add a case that is specifically property order dependent. Something like this:
```
class OrderMatters
{
    private readonly List<string> _list = new();

    public string? BindThisFirst { get; set; }

    public List<string> BindThisSecond => BindThisFirst is null ? throw new InvalidOperationException() : _list;
}
```

Do you think that would be valuable? Is property reflection order even well-defined in .NET Core (it wasn't in .NET Framework).

Reply:
I was hoping to test the case like 

```C#
public class Configuration
{
    public IEnumerable<int> FilteredConfigValues => ConfigValues.Where(p => p > 10);
}   public IEnumerable<int> ConfigValues { get; set; } = null!;
 ```

today this throws because `FilteredConfigValues` is initialized before `ConfigValues `. if the order is flipped like the following, it will not throw any more. 

```C#
public class Configuration
{
    public IEnumerable<int> ConfigValues { get; set; } = null!;
    public IEnumerable<int> FilteredConfigValues => ConfigValues.Where(p => p > 10);
}
```

Adding such a test case will validate the fix we are trying to do here.

Reply:
Added!